### DISCUSSION TITLE
Improving performance and understanding some detect.py params

### DISCUSSION THREAD

**Discussion by schubackindustries** [ultralytics/yolov5#4476] [Q&A]
Hey all, great library here, and thankfully I was able to get it up and running on Windows! Not always a guarantee these days... 

Anyways I was wondering about performance and some of the parameters I have available for detect.py. Specifically: 

1) What are the main parameters for improving real-time detection speed? 
An obvious one is "--imgsz" which I have tried reducing. 

2) What is "--max-det" aka "maximum detections per image"? 

3) Can I improve performance by only trying to detect a given subset of object classes? For example I'm mainly interested in "person" and "car". 

4) What is the difference between yolov5s.pt and yolov5s.ps ? Does one run faster than the other? 

Thanks for any help!

**Comment by glenn-jocher** [comment]
@schubackindustries you can speed up detection with smaller models, smaller image sizes, and batched inference using PyTorch Hub (see PyTorch Hub tutorial):

## YOLOv5 Tutorials

* [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://github.com/ultralytics/yolov5/wiki/Tips-for-Best-Training-Results)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://github.com/ultralytics/yolov5/issues/2518)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://github.com/ultralytics/yolov5/issues/475)
* [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://github.com/ultralytics/yolov5/issues/251) 🚀
* [Test-Time Augmentation (TTA)](https://github.com/ultralytics/yolov5/issues/303)
* [Model Ensembling](https://github.com/ultralytics/yolov5/issues/318)
* [Model Pruning/Sparsity](https://github.com/ultralytics/yolov5/issues/304)
* [Hyperparameter Evolution](https://github.com/ultralytics/yolov5/issues/607)
* [Transfer Learning with Frozen Layers](https://github.com/ultralytics/yolov5/issues/1314)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)
